ScanToShop/ViewControllers/RegisterViewController.m

```diff
@@ -94,42 +88,70 @@ - (UIImage *)resizeImage:(UIImage *)image withSize:(CGSize)size {
 }
 
 - (void)registerUser {
-    PFUser *newUser = [PFUser user];
     
     if ([self areUserInputFieldsEmpty]) {
-        [AlertManager loginAlert:@"" ViewController:self];
+        [AlertManager loginAlert:LoginErrorMissingInput ErrorString:nil ViewController:self];
+        return;
+    }
+    
+    if ([self fieldsContainSpacesOrNewlines]) {
+        [AlertManager loginAlert:SpaceNewlineError ErrorString:nil ViewController:self];
         return;
     }
     
-    newUser.username = self.usernameField.text;
-    newUser.password = self.passwordField.text;
-    newUser.email = self.emailField.text;
-    newUser[@"name"] = self.nameField.text;
-    newUser[@"image"] = [DatabaseManager getPFFileFromImage:self.profileImage.image];
+    if (![self containsValidPassword:self.passwordField.text]) {
```

Comment:
generally having to pass `self.something` as a parameter to an instance method is a sign you can just access that value in the instance method.